File: solutions/csharp/grains/1/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static ulong Square(int n) => 
        n < 1 || n > 64
            ? throw new ArgumentOutOfRangeException("The requested square must be between 1 and 64 (inclusive)")
            : Convert.ToUInt64(Math.Pow(2, n - 1));

    public static ulong Total() => 
        Enumerable.Range(1, 64)
            .Select(Square).Aggregate((accumulator, value) => accumulator + value);
}

File: solutions/scala/grains/1/src/main/scala/Grains.scala
object Grains {
  def square(value: Int):Option[BigInt] =
    if (value < 1 || value > 64) {
      None
    }
    else {
      Some(BigInt(2).pow(value - 1))
    }

  def total:BigInt = (1 to 64)
    .flatMap(square)
    .sum
}
